component/pyroscope/ebpf/ebpfspy/pprof.go

```diff
@@ -0,0 +1,99 @@
+//go:build linux
+
+package ebpfspy
+
+import (
+	"time"
+
+	"github.com/google/pprof/profile"
+	"github.com/prometheus/prometheus/model/labels"
+)
+
+type ProfileBuilders struct {
+	Builders   map[uint64]*ProfileBuilder
+	SampleRate int
+}
+
+func NewProfileBuilders(sampleRate int) *ProfileBuilders {
+	return &ProfileBuilders{Builders: make(map[uint64]*ProfileBuilder), SampleRate: sampleRate}
+}
+
+func (b ProfileBuilders) BuilderForTarget(hash uint64, labels labels.Labels) *ProfileBuilder {
+	res := b.Builders[hash]
+	if res != nil {
+		return res
+	}
+	builder := &ProfileBuilder{
+		locations: make(map[string]*profile.Location),
+		functions: make(map[string]*profile.Function),
+		Labels:    labels,
+		Profile: &profile.Profile{
+			Mapping: []*profile.Mapping{
+				{
+					ID: 1,
```

Comment:
Any way we can set the mapping filename based on the name of the binary ? On the database side we're using this to shard stacktraces.

Nothing urgent since we fallback to `service_name`

Reply:
Sure, we can add it. 
It may be a bit confusing, because there may be multiple executable and multiple mappings with same addresses, but different files because the pprof contains stacktraces from multiple processes.

Reply:
Ok let's start like this then we can experiment later.

Reply:
Don't change anything :)